Source/Core/Duality/Serialization/Serializer.cs

```diff
@@ -349,9 +355,16 @@ public void ReadObject<T>(out T obj)
 		public void WriteObject(object obj)
 		{
 			if (!this.CanWrite) throw new InvalidOperationException("Can't write object to a read-only serializer!");
-			this.BeginWriteOperation();
-			this.WriteObjectData(obj);
-			this.EndWriteOperation();
+
+			try
+			{
+				this.BeginWriteOperation();
+				this.WriteObjectData(obj);
```

Comment:
put the try finally here as well to make it consistent with the read part

Reply:
👍